### ISSUE TITLE
Odd Spacing and Line Breaks in Tutorial

### ISSUE THREAD

**Issue by elianna-b** [wordplaydev/wordplay#467]
### Expected Behavior
The tutorial is laid out as a conversation between mascots and the user to help the user gain familiarity with Wordplay. It begins with a short introduction and then prompts the user to attempt small tasks to learn about Wordplay.

- Ellipses are used to show trailing off or a continuation of a thought or idea to the next slide.
- The tutorial should be easily accessible and understandable for young students, with simple language and short sentences to avoid overwhelming the user learning code.

### Actual Behavior

**Ellipses Error:**
1. Go to [Silence 3-4/15](https://wordplay.dev/learn?locale=en-US&act=1&scene=1&pause=3&concept=function).
2. Read through the tutorial comment on the left.
3. Potential error: Notice the ellipses at the bottom that do not continue the text above.
   - The ellipses seem misplaced. They should indicate a continuation, but some pages lack them, and they have odd gaps away from words/ends of sentences. This could be a formatting error, or they might be unnecessary.

**Lack of Line Breaks Error:**
- Throughout the tutorial, most slides have few and short sentences with proper line breaks for better readability and to prevent overwhelming young students.
- However, some slides, for example, [Silence 9/15](https://wordplay.dev/learn?locale=en-US&act=1&scene=1&pause=9&concept=function), do not have any line breaks between questions.

**Questions:**
- Is there a reason there are no line breaks?
- Should we ensure consistent styling with line breaks for better readability?
- Could there be too much unnecessary plot/story in the tutorial that detracts from learning the content and causes confusion?

### Screenshots

**Ellipses Error:**
![Ellipses Error](https://github.com/wordplaydev/wordplay/assets/121900086/15173f3a-b9f5-4dd7-8224-74b5732c13ed)

**Lack of Line Breaks and Run-On Blurbs Examples:**
![Line Breaks Error 1](https://github.com/wordplaydev/wordplay/assets/121900086/e3063384-e1e4-4313-b72e-af96ce1385d4)
![Line Breaks Error 2](https://github.com/wordplaydev/wordplay/assets/121900086/8ae0cf92-21d1-43d4-812d-4a98c3d7868c)

### Environment
- **Laptop:** Windows 11, version 23H2
- **Browser:** Chrome Version 124.0.6367.203 (Official Build) (64-bit)

### Design Specification
_If this defect requires some redesign, include a design proposal here (and if you don't have one, add the `needs design` label, and you can just write TBD here). Design proposals should describe, in precise detail \_what_ the proposed design is, but not necessarily how it might be built. Note that the design proposal for a defect should align with the expected behavior you described above. Once the design is approved, we will add the `buildable` tag is added.

**Comment by elianna-b** [comment]
Preface: I am working with a team to create a more accessible tutorial with a better user experience. I originally posted these questions on Discord but transferred them to individual issue postings because Discord is too ephemeral. I am seeking to understand the original design decisions and why certain features were created as they were, or if they were unintentional. If any features are unintended or could be improved, I am happy to address the short issues I posted.

**Comment by amyjko** [comment]
Thanks for reporting @elianna-b! 

All of the issues you noted in this issue are content questions, not layout issues or bugs. For example, the "…" is intentionally a new paragraph, indicating that the character is pausing and saying nothing. Paragraphs with multiple sentences are intentional, in that some characters speak verbosely and monologue without pausing.

What you're really critiquing here is the dialog. And that opens a very important question: what makes a _good_ tutorial? What kind of dialog _should_ a tutorial have? Was my original idea of creating a world where glyphs had personality and different communication styles a good one, and for whom? Should there be tutorials that are _only_ focused on content, and stripped of any narrative? There are no easy answers to these questions: portraying this tutorial as a multi-act play with resonate with some youth, and not with others. The question is who the tutorial is serving, who it is not, and who should be served.

I get from your questions that you might have opinions about these questions. Share them here, and let's discuss!

**Comment by elianna-b** [comment]
Hi @amyjko, thank you for your response!

I would love to discuss these content questions, and I have some opinions. I have also read through the WCAG (although not an expert) and have experience with UX and design work. Here are my thoughts based on that information:

1. I believe the "…" is counterintuitive. Clicking the arrow for the next slide as a pause or new paragraph is a common practice in Wordplay. This inconsistency throughout the tutorial could confuse youth unfamiliar with this method.

2. The dialogue should be at a middle/high school reading level. Feedback mentioned prior suggests it currently sounds written for and by a college audience.

3. Creating a world where glyphs have personality and different communication styles is engaging for younger students. For college students, it could seem tedious and disengaging, but this is not our demographic, and younger students might benefit from connecting with the program through characters, making Wordplay's theme memorable, unique, and easily recognizable which is important for an application's identity. The challenge is balancing thematic elements. While the core tutorial handles this well, run-on paragraphs without line breaks and excessive focus on the fictional world can detract from the tutorial's helpful aspects.

4. A verbose, monologue-style personality can be detrimental. It is not recognized as a personality by screen readers, and there is no audio for tone. Pauses (line breaks) improve accessibility, making the text easier to discern for those with eyesight issues or difficulties reading long paragraphs with small fonts and a lack of spacing.

5. I considered the idea of different tutorials, with one focused on content and stripped of narrative elements. Making the platform more customizable, such as allowing users to skip sections or slides, or checking a box to remove the narrative, could be beneficial. I like this feature: ![image](https://github.com/wordplaydev/wordplay/assets/121900086/96bddf28-2246-4a21-aed9-3068366c251f))

   However, the section names seem vague, making it difficult to determine their content at a glance. Some sections seem based on the narrative rather than programming techniques you are learning. What is more important for the user? What is more engaging or easily understandable for our demographic? Programming terminology can be confusing or overwhelming, how can we find a balance?

I agree that there is no single right answer. Instead of creating a whole new tutorial or separate ones, we should build upon and improve this, until receiving iterative feedback from test users.

Understanding what makes a good tutorial might be out of my scope, unfortunately, but it would be interesting to ask the teachers and students we're collaborating with this summer and research this topic.

Please let me know what you think!

**Comment by amyjko** [comment]
Thanks for brainstorming @elianna-b. Here are a few ways forward:

* Submit an issue proposing to avoid use of "…" to mean "trailing pause at the end of a sentence". We can have that debate there; it is a convention in English writing, but also a convention in user interfaces, drawn from English writing, and so it can mean two different things. That's not necessarily bad, but it's worth examining the tradeoffs.

* Issue #460 already addresses reading levels. If you have more to add on that topic, please comment there. You can see some of the technical challenges of enforcing this across languages.

* Submit an issue proposing to create a new tutorial, or a version of the current tutorial, devoid of narrative. As part of that, share your arguments why you believe some middle school students would resonate more with content that is strictly informational, and how we might manage two different styles of tutorials across many natural languages we hope to support (e.g., does every scene have two versions, one narrative, and one not?)

* Close this issue, as its scope is too big.

**Comment by elianna-b** [comment]
Added on to Issue [#460](https://github.com/wordplaydev/wordplay/issues/460) and closed this seperate discussion.
